### PR TITLE
[Platforms] Refactor xpu code

### PR DESCRIPTION
**Pull request by MengqingCao**
Part of #9268 

**Mainly changes**:
Refactor `_verify_and_get_xxx_config` to `XPUPlatform.check_and_update_config`

### CODE REVIEW

**Comment by MengqingCao** [comment]
cc @youkaichao

File: vllm/platforms/cpu.py [hunk 1]
```diff
@@ -62,12 +62,12 @@ def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
 
         if kv_cache_space >= 0:
             if kv_cache_space == 0:
-                cache_config.cpu_kvcache_space_bytes = 4 * GiB_bytes  # type: ignore
```
Comment (DarkLight1337) [inline vllm/platforms/cpu.py:65]: This particular field is being used in cpu worker.

File: vllm/platforms/xpu.py [hunk 2]
```diff
@@ -34,3 +41,27 @@ def get_device_total_memory(cls, device_id: int = 0) -> int:
...
+
+        # check and update parallel config
+        parallel_config = vllm_config.parallel_config
+        if (parallel_config.distributed_executor_backend is not None
+                and parallel_config.distributed_executor_backend != "ray"):
+            logger.warning(
+                "%s is not supported on XPU, fallback to ray distributed"
+                " executor backend.",
+                parallel_config.distributed_executor_backend)
+            parallel_config.distributed_executor_backend = "ray"
```
Comment (youkaichao) [inline vllm/platforms/xpu.py:67]: I think this part of code is not used right now. we can directly delete them

File: vllm/platforms/cpu.py (hunk 1 above)
Comment (MengqingCao) [reply vllm/platforms/cpu.py:65]: Ok, I'll revert this then, thanks!

[the remark this answers is not in the document]
Comment (MengqingCao) [reply vllm/platforms/cpu.py:65]: I'll revert this and make this pr clean.

File: vllm/platforms/xpu.py (hunk 2 above)
Comment (MengqingCao) [reply vllm/platforms/xpu.py]: Done

**Review by youkaichao — APPROVED** [review]
thanks for cleaning it up!

File: vllm/platforms/xpu.py [hunk 3]
```diff
@@ -34,3 +41,17 @@ def get_device_total_memory(cls, device_id: int = 0) -> int:
...
+        model_config = vllm_config.model_config
+        if model_config.dtype == torch.bfloat16:
+            logger.warning(
+                "bfloat16 is not fully supported on XPU, casting to float16.")
+            model_config.dtype = torch.float16
+        if not model_config.enforce_eager:
+            logger.warning(
+                "CUDA graph is not supported on XPU, fallback to the eager "
+                "mode.")
+            model_config.enforce_eager = True
```
Comment (yma11) [inline vllm/platforms/xpu.py:57]: seems lost the original `parallel_config` check?
```
    if (config.distributed_executor_backend is not None
            and config.distributed_executor_backend != "ray"):
        logger.warning(
            "%s is not supported on XPU, fallback to ray distributed executor "
            "backend.", config.distributed_executor_backend)
        config.distributed_executor_backend = "ray"
```

Comment (youkaichao) [reply vllm/platforms/xpu.py:57]: I think that function `_verify_and_get_parallel_config` is not called.

Comment (yma11) [reply vllm/platforms/xpu.py:57]: This check should be included for xpu platform. The call stack seems get broken in [commit](https://github.com/vllm-project/vllm/commit/18bd7587b78b3b9868fea29d59ae8c3600c3e5a5#diff-d1c5ec4ddd588e3b7cac13bde85a98ac5b20686dc16b9da3b1c324467c3be2b5).

Comment (youkaichao) [reply vllm/platforms/xpu.py:57]: oh sorry for that. can you help review https://github.com/vllm-project/vllm/pull/10479 ?

**Pull request merged**